### PR TITLE
MQTT OTA Update review

### PR DESCRIPTION
**Pull request by mikee47**

### CODE REVIEW

**Comment by mikee47** [comment]
One thing which keeps cropping up is use of casts with simple types. I avoid use of 'C' style casts, such as `(uint8_t)(c)` as the function-style casts are cleaner `uint8_t(c)`. The preferred approach would be `uint8_t{c}` but gives 'narrowing' warning/error in some situations. Using `static_cast<uint8_t>(c)` is probably the best way but is messier and seems a bit overkill for this sort of thing where no pointers are involved.

See https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es48-avoid-casts

**Comment by slaff** [comment]
@mikee47 And thanks a lot for your reviews!

File: Sming/Libraries/OtaUpgradeMqtt/tools/deployer/app/application.cpp [hunk 1]
```diff
@@ -45,61 +45,68 @@ template <typename T> void println(const T& arg)
...
-	if(output == nullptr) {
-		return -1;
-	}
-
-	int written = 0;
+	size_t written = 0;
 	if(useVarInt) {
 		while(patchVersion > 0x7f) {
-			if(output->write(((uint8_t)(patchVersion)) | 0x80) < 0) {
+			if(output.write(uint8_t(patchVersion | 0x80)) == 0) {
```
[the remark this answers is not in the document]
Comment (mikee47) [reply Sming/Libraries/OtaUpgradeMqtt/tools/deployer/app/application.cpp:53]: It was the `<0` check which alerted me as the method is `size_t Print::write(uint8_t)`. Agreed explicit check is best.

[the remark this answers is not in the document]
Comment (mikee47) [reply Sming/Libraries/OtaUpgradeMqtt/tools/deployer/app/application.cpp:53]: The compiler should have complained about `<0`...

**Pull request merged**